src/pywifes/splice.py

```diff
@@ -0,0 +1,330 @@
+        flux[overlap] = (
+            flux_B[overlap] / fluxvar_B[overlap] + flux_R[overlap] / fluxvar_R[overlap]
+        ) * fluxVar[overlap]
+
+        flux[redOnly] = flux_R[redOnly]
+        fluxVar[redOnly] = fluxvar_R[redOnly]
+
+        return flux, fluxVar
+
+
+def splice_spectra(blue_spec_path, red_spec_path, output):
+    """
+    The main routine
+    """
+    # Create to instances of SingleSpec
+    blueSpec = SingleSpec(blue_spec_path)
+    redSpec = SingleSpec(red_spec_path)
+
+    # Join the spectra
+    print("Splicing spectra")
+    flux, fluxVar = join_spectra(blueSpec, redSpec)
+
+    if len(flux) == 1 :
+        print("No spectral overlap")
+
+    else:
+        # Write out the results
+        # Use the header in red arm to start with
+        # Add additional blue CCD keywords as required - pending
+        hdulist = fits.HDUList(fits.PrimaryHDU())
+        hdulist[0].data = flux
+        hdulist[0].header = redSpec.header
+        hdulist[0].header["CRPIX1"] = blueSpec.header["CRPIX1"]
+        hdulist[0].header["CRVAL1"] = blueSpec.header["CRVAL1"]
+        hdulist[0].header["CDELT1"] = blueSpec.header["CDELT1"]
+        hdulist[0].header["CTYPE1"] = "WAVE"
+        hdulist[0].header["CUNIT1"] = "Angstrom"
+
+        hdr_fluxvar = fits.Header()
+        hdr_fluxvar["EXTNAME"] = "VARIANCE"
+        hdr_fluxvar["CRPIX1"] = blueSpec.header["CRPIX1"]
+        hdr_fluxvar["CRVAL1"] = blueSpec.header["CRVAL1"]
+        hdr_fluxvar["CDELT1"] = blueSpec.header["CDELT1"]
+        hdr_fluxvar["CTYPE1"] = "WAVE"
+        hdr_fluxvar["CUNIT1"] = "Angstrom"
+        hdr_fluxvar["BUNIT"] = "(count / Angstrom)^2"
+
+        hdu_fluxvar = fits.ImageHDU(data=fluxVar, header=hdr_fluxvar)
+        hdulist.append(hdu_fluxvar)
+
+        print("Saving spliced spectra")
+        hdulist.writeto(output, overwrite=True)
+        hdulist.close()
+
+
+
+
+def join_cubes(blue_path, red_path):
+
+    # Read red data and metadata
+    red_flux_cube = fits.getdata(red_path, 0, header=False)
+    red_fluxvar_cube = fits.getdata(red_path, 1, header=False)
+
+    redCRVAL1 = fits.getheader(red_path)["CRVAL3"]
+    redCDELT1 = fits.getheader(red_path)["CDELT3"]
+    redCRPIX1 = fits.getheader(red_path)["CRPIX3"]
+    redNAXIS1 = fits.getheader(red_path)["NAXIS3"]
+    red_wl = (np.arange(redNAXIS1) - redCRPIX1 + 1) * redCDELT1 + redCRVAL1
+    red_minWL = min(red_wl)
+    red_maxWL = max(red_wl)
+
+    # Read blue data and metadata
+    blue_flux_cube = fits.getdata(blue_path, 0, header=False)
+    blue_fluxvar_cube = fits.getdata(blue_path, 1, header=False)
+    blueCRVAL1 = fits.getheader(blue_path)["CRVAL3"]
+    blueCDELT1 = fits.getheader(blue_path)["CDELT3"]
+    blueCRPIX1 = fits.getheader(blue_path)["CRPIX3"]
+    blueNAXIS1 = fits.getheader(blue_path)["NAXIS3"]
+    blue_wl = (np.arange(blueNAXIS1) - blueCRPIX1 + 1) * blueCDELT1 + blueCRVAL1
+    blue_minWL = min(blue_wl)
+    blue_maxWL = max(blue_wl)
```

Comment:
Instead of a function, I read the header into a variable and then just take the corresponding keys. I think that, even when is a bit repetitive, we this structure and the new variable name you suggested we have it now pretty clear.